packages/eslint-rules/src/rules/prefer_amplify_errors.ts

```diff
@@ -0,0 +1,46 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+import { join, sep } from 'path';
+
+export const preferAmplifyErrorsRule = ESLintUtils.RuleCreator.withoutDocs({
+  create(context) {
+    return {
+      // This naming comes from @typescript-eslint/utils types.
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      NewExpression(node) {
+        const checkNode = (errorType: string, ignorePatterns: string) => {
+          const fileNameWithPath = context.physicalFilename!.replace(
+            join(context.cwd, sep),
+            ''
+          );
+
+          const ignoreMatch = [
+            ...fileNameWithPath.matchAll(new RegExp(ignorePatterns, 'g')),
+          ];
```

Comment:
I thought we abandon the regex route.

Reply:
I left it for ignoring tests, since I would otherwise have to include a long list of ignores for every directory level tests could be on. I can experiment with removing regex and moving everything to the index.ts file

Reply:
I see. It would be best if configuration of what files are included and excluded is in one place.
I.e. in that `overrides` section in the other file.

Things to try:
1. The `files` in `overrides` takes [glob](https://en.wikipedia.org/wiki/Glob_(programming)) . Perhaps a pattern like `packages/cli/src/**/*!(.test.ts)` could work (I made the pattern up, but there might be some option to negate group of characters.
2. If above doesn't work (and maybe it's more explicit) the `overrides` should also take `ignores ` (see [docs](https://eslint.org/docs/latest/use/configure/configuration-files)) which is also a "glob" but is used to filter out unwanted files.

If these two options don't yield anything good, we can keep regex.

Reply:
the `excludedFiles` field in `overrides` actually takes care of this now, turns out if you use `**/*.test.ts` it takes care of ignoring test files as all levels (which I did not realize previously). So the regex is now gone.